tests/test_host/test_external_repo.py

```diff
@@ -9,20 +9,20 @@
 
 class ExternalRepoTest(unittest.TestCase):
     def test_download(self):
-        host_paths = paths.host()
+        external_repo_dir = pathlib.Path(paths.host().external_repo_dir)
         repo = "twosixlabs/armory-example"
         repo_name = repo.split("/")[-1]
 
-        download_and_extract_repo(repo)
-        self.assertTrue(os.path.exists(f"{host_paths.external_repo_dir}/{repo_name}"))
-        self.assertTrue(
-            os.path.isfile(
-                str(
-                    pathlib.Path(
-                        f"{host_paths.external_repo_dir}/{repo_name}/README.md"
-                    )
-                )
-            )
-        )
+        download_and_extract_repo(repo, external_repo_dir=external_repo_dir)
+        basedir = external_repo_dir / repo_name
 
-        shutil.rmtree(str(pathlib.Path(f"{host_paths.external_repo_dir}/{repo_name}")))
+        self.assertTrue(os.path.exists(basedir))
+        self.assertTrue(os.path.isfile(basedir / "README.md"))
+        shutil.rmtree(basedir)
+        os.remove(external_repo_dir / (repo_name + ".tar.gz"))
+        try:
+            os.rmdir(external_repo_dir)
+        except OSError:
+            # Only delete if empty
+            pass
```

Comment:
Could you elaborate on why were doing this?

Reply:
The removal of the external_repo_dir is necessary for overall cleanup. This essentially ensures that all that is in the external_repo_dir is the tar file and the extracted repo. If there are additional files or folders, the end assertion will fail.